src/Http/Request/Body/ContentNegotiationRequestBodyParser.php

```diff
@@ -8,14 +8,28 @@
 
 final class ContentNegotiationRequestBodyParser implements RequestBodyParser
 {
-    public function parse(ServerRequestInterface $request): array
+    private RequestBodyParser $jsonRequestBodyParser;
+
+    public function __construct()
+    {
+        $this->jsonRequestBodyParser = new JsonRequestBodyParser();
+    }
+
+    public function withJsonRequestBodyParser(RequestBodyParser $jsonRequestBodyParser)
```

Comment:
Maybe remove this method and add a required parameter to the constructor? Then the caller decides to construct a `JsonRequestBodyParser` on the fly or for example use the new `JsonSchemaValidatingRequestBodyParser`

Reply:
I'm just going to remove this `ContentNegotiationRequestBodyParser` because the more I think about it, the less there seems to be a need for it:

- We currently only support JSON
- If we want to support other formats in the future, we can for example convert them to JSON first when they enter the application and then we don't need to support multiple formats in each controller / request handler

Reply:
Removed!